README.md

```diff
@@ -1,26 +1,341 @@
+The ultimate goal of the release toolkit is to allow for fully automated release pipelines that produce meaningful changelogs and respect semver standards. For this reason, the recommended way of implementing the toolkit is as a scheduled job that runs on the `main`/`master` branch of your source tree, for example every tuesday at midnight.
 
+The release toolkit allows you to build a pipeline that creates a release with its changelog and version automatically if there are any unreleased changes in your main branch.
+
+However, it is possible that, in some cases, you may want to hold changes from being released as you want to "pack" them with future changes for UX reasons. To this end, the release toolkit includes the concept of holding releases. A release can be held by including a `## Held` L2 header in `CHANGELOG.md`. This header must be followed by a paragraph explaining why automated releases are on hold:
+
+```markdown
+# Changelog
+This is based on blah blah blah
+
+## Held
+
+Holding release as it contains massive breaking changes
+
+## Unreleased
+
+### Breaking
+- Support has been removed
+
+## v1.2.3 - 20YY-DD-MM
+```
+
+The presence of this header does not modify the fundamental behavior of any command. It does, however, cause `generate-changelog` to include a `held: true` property in `changelog.yaml`:
+
+```yaml
+held: true
+changes:
+  - type: breaking
+    message: Support has been removed
+```
+
+This flag can be readily checked using the `is-held` action/command. `is-held` will echo the value of this property to `stdout`, and can be configured to exit with non-zero if `held` is set to `true`. While running as an action, it will also set the `is-held` step output to the value of the property. This allows your pipeline to react to this property and skip the release process if a `## Held` entry is present in your `CHANGELOG.md`.
+
+`update-changelog` will not behave any different in presence of a `## Held` header, and will remove it before integrating the changelog for the latest release.
+
+### Full example of automated releasing pipeline
+
+![release-toolkit.png](release-toolkit.png)
+
+```yaml
+// TODO
+```
+
+---
+
+### Full architecture overview
+# Changelog YAML
+
+The different release-toolkit components, either can be executed independently, or need to be fed with a custom-format yaml file from where all the current release changes, dependencies and notes will be computed.
+
+## User-facing changelog
+
+The release toolkit is intended to read entries from a CHANGELOG.md following the [keep a changelog format](https://keepachangelog.com/). 
+
+Any change or note, the user wants to add to the next release, should be placed under the Level2 header `Unreleased`.
+
+The `generate-yaml` command will read the entries in the `Unreleased` section, adding all the correct entries to the changelog.yaml file. We describe the types of entries in the next section.
+
+
+
+## CLI
```

Comment:
This readme is very lengthy already, I wonder if we should move all this to a separate `.md` file for clarity. E.g. `README_cli.md`.